### PR TITLE
Updated devtools-guide-chromium/whats-new/2021/04/devtools file

### PR DESCRIPTION
**Pull request by jm-trd-ms**
Updated content in devtools-guide-chromium/whats-new/2021/04/devtools.md.

### CODE REVIEW

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md [hunk 1]
```diff
@@ -166,17 +166,215 @@ The [Microsoft Edge Tools for Visual Studio Code extension][VisualstudioMarketpl
...
+
+#### Enhanced support for breakpoints debugging  
+
+In Microsoft Edge version 90 or earlier, DevTools only set breakpoints on one raw location.  In Microsoft Edge version 91 or later, DevTools properly set breakpoints in multiple scripts.  This feature is a modern support code splitting feature.  Multiple routes may include one shared component and are named code splits.  JavaScript bundlers include [Webpack][WebpackJsMain], and [Rollup][RollupjsMain].  In Microsoft Edge version 91 or later, DevTools sets breakpoints in all relevant locations correctly.  To review the history of this feature in the Chromium open-source project, navigate to Issues [1142705][CR1142705], [979000][CR979000], and [1180794][CR1180794].  
+
+#### Support hover preview with [] notation  
+
+DevTools now support hover preview on JavaScript member expressions that use the `[]` notation in the **Sources** tool.  To review the history of this feature in the Chromium open-source project, navigate to Issue [1178305][CR1178305].  
+
+<!--
```
Comment (zoherghadyali) [inline microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:337]: For this image, navigate to: https://codepen.io/zoherghadyali/full/jOBOBBX. Open DevTools > **Sources**. Select `Ctrl`+`P` and open **pen.js**. Set a breakpoint on line 4 and refresh the page. Hover over `arr[i]` and you should see the hover display the correct value of `a` in the array.

Comment (zoherghadyali) [reply microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:337]: Here's what I got: 
image

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md [hunk 2]
```diff
@@ -166,17 +166,215 @@ The [Microsoft Edge Tools for Visual Studio Code extension][VisualstudioMarketpl
...
+:::image type="complex" source="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/XFxlQjvLVWJUiV24zDw2.msft.png" alt-text="Support hover preview with [] notation" lightbox="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/XFxlQjvLVWJUiV24zDw2.msft.png":::
+   Support hover preview with `[]` notation  
+:::image-end:::  
+-->  
+
+#### Improved outline of HTML files  
+
+DevTools now has better outline support for `.html` files.  In the **Sources** tool, open the `.html` file.  To turn on \(or off\) the code outline, select `Ctrl`+`Shift`+`O` on Windows/Linux or `Cmd`+`Shift`+`O` on macOS.  In the following figure, DevTools now correctly list all functions in the outline.  Previously, DevTools only displayed some of the functions.  To review the history of this feature in the Chromium open-source project, navigate to Issues [761019][CR761019] and [1191465][CR1191465].  
+
+:::image type="complex" source="../../media/2021/04/sources-pages-html-code-outline.msft.png" alt-text=" Improved outline of HTML files" lightbox="../../media/2021/04/sources-pages-html-code-outline.msft.png":::
```
Comment (zoherghadyali) [inline microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:347]: This image doesn't really show the new functionality. I would navigate to: https://codepen.io/zoherghadyali/full/jOBOBBX and open DevTools > **Sources**. Open the jOBOBBX file under `codepen.io/zoherghadyali/full`. You should see a file that looks HTML. Now select `Ctrl`+`P` and enter `@`. Here's what I got: 
image

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md [hunk 3]
```diff
@@ -166,17 +166,215 @@ The [Microsoft Edge Tools for Visual Studio Code extension][VisualstudioMarketpl
...
+   Improved outline of HTML files  
+:::image-end:::  
+
+#### Proper error stack traces for Wasm debugging  
+
+In Microsoft Edge version 90 or earlier, DevTools only displayed generic Wasm references in the Error stack traces.  
+In Microsoft Edge version 91 or later, DevTools resolves inline function requests and displays proper error stack traces for Wasm debugging.  
+
+:::row:::
```
Comment (zoherghadyali) [inline microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:356]: For this image, navigate to: https://emscripten-dbg-stories.netlify.app/inlining-dwarf.html. Open DevTools > **Sources**. Select `Ctrl`+`P` and enter `inlining-dwarf.wasm`. Set a breakpoint at `0x0028a`. Refresh the page. When the breakpoint is hit, switch to the **Console** and then enter `new Error` and run that. The error now contains the correct stack trace. Here's what I got: 


image

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md [hunk 4]
```diff
@@ -166,17 +166,215 @@ The [Microsoft Edge Tools for Visual Studio Code extension][VisualstudioMarketpl
...
+
+DevTools now has better outline support for `.html` files.  In the **Sources** tool, open the `.html` file.  To turn on \(or off\) the code outline, select `Ctrl`+`Shift`+`O` on Windows/Linux or `Cmd`+`Shift`+`O` on macOS.  In the following figure, DevTools now correctly list all functions in the outline.  Previously, DevTools only displayed some of the functions.  To review the history of this feature in the Chromium open-source project, navigate to Issues [761019][CR761019] and [1191465][CR1191465].  
+
+:::image type="complex" source="../../media/2021/04/sources-pages-html-code-outline.msft.png" alt-text=" Improved outline of HTML files" lightbox="../../media/2021/04/sources-pages-html-code-outline.msft.png":::
+   Improved outline of HTML files  
+:::image-end:::  
+
+#### Proper error stack traces for Wasm debugging  
+
+In Microsoft Edge version 90 or earlier, DevTools only displayed generic Wasm references in the Error stack traces.
```
Comment (zoherghadyali) [inline microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:353]:
```suggestion
In Microsoft Edge version 90 or earlier, DevTools only displayed generic Wasm references in Error stack traces.  In Microsoft Edge version 91 or later, DevTools resolves inline function requests and displays the source location in Error stack traces for Wasm debugging.  To learn more about Error stack traces in the **Console**, navigate to [error | Console API reference](https://docs.microsoft.com/microsoft-edge/devtools-guide-chromium/console/api#error).  
```

**Review by zoherghadyali — CHANGES_REQUESTED** [review]

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md [hunk 5]
```diff
@@ -166,17 +166,215 @@ The [Microsoft Edge Tools for Visual Studio Code extension][VisualstudioMarketpl
...
+#### Support for new JavaScript features  
+
+DevTools now support the new [Private brand checks a.k.a. #foo in obj][V8DevFeaturesPrivateBrandChecks] JavaScript language feature, a.k.a `#foo in obj`.  The private brand checks feature extends the [in operator][MdnDocsWebJavascriptReferenceOperatorsIn] to support the [Private class fields][V8DevFeaturesClassFieldsPrivateClassFields] to test on a specific object.  Try it in the **Console** and **Sources** tools.  Also, to inspect the private fields, complete the following actions.  
+
+1.  Navigate to **debugger** pane.  
+1.  Navigate to the **Scope** section.  
+    
+To review the history of this feature in the Chromium open-source project, navigate to Issue [11374][CR11374].  
+
+<!--
```
[the remark this answers is not in the document]
Comment (jm-trd-ms) [reply microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:342]: Updated.

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md (hunk 1 above)
Comment (jm-trd-ms) [reply microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:359]: Updated.

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md (hunk 2 above)
Comment (jm-trd-ms) [reply microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:369]: Updated.

File: microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md (hunk 3 above)
Comment (jm-trd-ms) [reply microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md:381]: Updated.

**Review by zoherghadyali — APPROVED** [review]

**Pull request merged**